kornia/enhance/jpeg.py

```diff
@@ -503,11 +503,32 @@ def jpeg_codec_differentiable(
     # Check value range of JPEG quality
     KORNIA_CHECK(
         (jpeg_quality.amin().item() >= 0.0) and (jpeg_quality.amax().item() <= 100.0),
-        f"JPEG quality is out of range. Expected range is [0, 99], "
+        f"JPEG quality is out of range. Expected range is [0, 100], "
         f"got [{jpeg_quality.amin().item()}, {jpeg_quality.amax().item()}]. Consider clipping jpeg_quality.",
     )
-    # Get original shape
-    H, W = image_rgb.shape[2:]
+    # Save original shape
+    original_shape = image_rgb.shape
+    # Reshape to [B, 3, H, W]
+    image_rgb = image_rgb.reshape(-1, 3, *original_shape[-2:])
```

Comment:
one alternative here is to reuse a decorator we have in utils `@perform_keep_shape_image` -- you can see an example for the resize function: https://github.com/kornia/kornia/blob/main/kornia/geometry/transform/affwarp.py#L525

we haven't been using it very much across of the library because its lack of support to torchscript, but with the recent version of pytorch this seems to be sustainable improved. /cc @johnnv1 to discuss about adopting it more

Reply:
Using the `perform_keep_shape_image` will make it incompatible with jit (kinda going against the onnx improvements) -- but it looks to be supported by the torch compile, which will replace jit maybe in a year

So maybe we can go with or support jit or supporting compile, with a preference for compile? https://pytorch.org/docs/stable/export.html#existing-frameworks

Reply:
Used the `perform_keep_shape_image` decorator. I guess `torch.compile` support might be more useful. Thanks for the feedback :)

Reply:
cross-linking to https://github.com/kornia/kornia/issues/2559